bash-preexec.sh

```diff
@@ -47,7 +47,17 @@ __bp_last_argument_prev_command="$_"
 
 # Command to set our preexec trap. It's invoked once via
 # PROMPT_COMMAND and then removed.
-__bp_trap_install_string="trap '__bp_preexec_invoke_exec \"\$_\"' DEBUG;"
+__bp_trap_install() {
+  local prior_trap=$(trap -p DEBUG | sed "s/[^']*'\(.*\)'[^']*/\1/")
```

Comment:
That's really interesting. I think this is due to `set -T`, I'll experiment some more but I may have to go back to something like your `__bp_trap_install_string` solution if `set -T` won't work. I'll get back to you.